File: Algorithms/p049_Group_Anagrams/p049_Group_Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> hash_table;
        for(auto &s : strs){
            string s1(s);
            sort(s1.begin(),s1.end());
            hash_table[s1].push_back(s);
        }
        vector<vector<string>> res;
        for(auto &e : hash_table){
            res.push_back(move(e.second));
        }
        return res;
    }
};

File: Algorithms/p049_Group_Anagrams/p049_Group_Anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        hash_table = collections.defaultdict(list)
        for s in strs:
            hash_table[''.join(sorted(s))].append(s)
        res = []
        for key in hash_table:
            res.append(hash_table[key])
        return res
